### DISCUSSION TITLE
How to get data from ETCD?

### DISCUSSION THREAD

**Discussion by ameryhan** [apache/apisix#6004] [General]
怎么获取在etcd里面的某个目录全部数据

尝试过的方法1
  local res, err = core.etcd:getkey('/config_path')
 报错信息如下
![image](https://user-images.githubusercontent.com/2375525/148041696-0583dee7-b927-4851-99eb-0cc4c75c08cb.png)

尝试过的方法2
  local res, err = core.config:getkey('/config_path')
返回的数据为空。

请问用那种方法可以做呢？

**Comment by tokers** [comment]
Use the `readdir` method in lua-resty-etcd.

**Comment by juzhiyuan** [comment]
related issue https://github.com/apache/apisix/issues/6005
